manim/scene/scene_file_writer.py

```diff
@@ -388,15 +381,17 @@ def write_frame(self, frame_or_renderer: np.ndarray | OpenGLRenderer):
         elif config.renderer == RendererType.CAIRO:
             frame = frame_or_renderer
             if write_to_movie():
-                self.writing_process.stdin.write(frame.tobytes())
+                av_frame = av.VideoFrame.from_ndarray(frame, format="rgba")
+                for packet in self.video_stream.encode(av_frame):
+                    self.video_container.mux(packet)
             if is_png_format() and not config["dry_run"]:
                 self.output_image_from_array(frame)
 
     def write_opengl_frame(self, renderer: OpenGLRenderer):
         if write_to_movie():
-            self.writing_process.stdin.write(
-                renderer.get_raw_frame_buffer_object_data(),
-            )
+            av_frame = av.VideoFrame.from_ndarray(renderer.get_frame(), format="rgba")
```

Comment:
I think we can apply the `-vf vlip` filter directly on the ndarray object, since it just flips one axis.

Reply:
After checking, seems like `get_frame()` already flips the vertical axis with `np.flipud`:

https://github.com/ManimCommunity/manim/blob/40766cb648de05941694bacb773812903db2a817/manim/renderer/opengl_renderer.py#L559-L566